integrations/nvidia/src/haystack_integrations/utils/nvidia/utils.py

```diff
@@ -44,8 +46,80 @@ def url_validation(api_url: str, default_api_url: Optional[str], allowed_paths:
 
 
 def is_hosted(api_url: str):
-    """"""
+    """Check if the api_url belongs to api catalogue."""
     return urlparse(api_url).netloc in [
         "integrate.api.nvidia.com",
         "ai.api.nvidia.com",
     ]
+
+
+def lookup_model(name: str) -> Optional[Model]:
+    """
+    Lookup a model by name, using only the table of known models.
+    The name is either:
+        - directly in the table
+        - an alias in the table
+        - not found (None)
+    Callers can check to see if the name was an alias by
+    comparing the result's id field to the name they provided.
+    """
+    model = None
+    if not (model := MODEL_TABLE.get(name)):
+        for mdl in MODEL_TABLE.values():
+            if mdl.aliases and name in mdl.aliases:
+                model = mdl
+                break
+    return model
+
+
+def determine_model(name: str) -> Optional[Model]:
+    """
+    Determine the model to use based on a name, using
+    only the table of known models.
+
+    Raise a warning if the model is found to be
+    an alias of a known model.
+
+    If the model is not found, return None.
+    """
+    if model := lookup_model(name):
+        # all aliases are deprecated
+        if model.id != name:
+            warn_msg = f"Model {name} is deprecated. Using {model.id} instead."
+            warnings.warn(warn_msg, UserWarning, stacklevel=1)
+    return model
+
+
+def validate_hosted_model(class_name: str, model_name: str, client) -> None:
+    """
+    Validates compatibility of the hosted model with the client.
+
+    Args:
+        model_name (str): The name of the model.
+
+    Raises:
+        ValueError: If the model is incompatible with the client.
+    """
+    if model := determine_model(model_name):
+        if not model.client:
+            warn_msg = f"Unable to determine validity of {model.id}"
+            warnings.warn(warn_msg, stacklevel=1)
+        elif model.model_type == "embedding" and class_name in ["NvidiaTextEmbedder", "NvidiaDocumentEmbedder"]:
```

Comment:
If this is a valid condition maybe we can remove it? and this function would only test invalid conditions and raise and ValueError on such conditions occur?